woocommerce/woocommerce-gateway-stripe
# Migrate WCS Stripe Oauth code to Stripe

This is a work in progress for moving WCS Stripe Connect Oauth code into Stripe.

### Changes proposed in this Pull Request:

Add the following from WCS:

- Connect api
- Base REST controller
- REST deauthorize controller
- REST oauth-init controller
- REST oauth-connect controller

I've also removed the Jetpack dependency from Oauth requests, so both Jetpack and WCS are no longer necessary for the OAuth flow to work.

### Testing

The changes in this PR are dependent upon the following two PRs in [woocommerce-admin](https://github.com/woocommerce/woocommerce-admin/pull/4933) and [woocommerce-connect-server](https://github.com/Automattic/woocommerce-connect-server/pull/1443)

To test these changes now, you will need to have both of the above installed/running locally. Otherwise, you can wait for the Connect Server PR to be merged as the changes there won't impact the current OAuth flow.

To test:

- Get a fresh install of WP set up on a test site and install WooCommerce
- With the WC-Admin changes above checked out, kick off the setup wizard
- Skip the WCS/Jetpack install step
- Complete the Stripe Payment task. With these changes, the oauth flow should work here even though Jetpack and WCS are not installed.
- Go to the Stripe settings page
- Look for the **Reset Stripe account keys** button and click it: https://d.pr/i/njErdY
- The page should reload and your keys should no longer be saved.
- Look for the **Setup or link an existing Stripe Account** button and click it: https://d.pr/i/G2fAkt
- This should kick off the OAuth flow again and re-save your Stripe keys

Note: When testing with a sandbox Stripe account, you will need to manually change instances of publishable_key and secret_key to test_publishable_key and test_secret_key in the following WC-Admin files:

    client/task-list/tasks/payments/stripe.js
    client/task-list/tasks/payments/methods.js


### Questions

I wasn't sure about a few pieces of the OAuth code so just copied these over anyway to be safe.

- Is adding "useful data" to the OAuth request necessary in `WC_Stripe_Connect_API`?
- Are the deauth controller and related methods necessary? How are these used and are they still relevant now that Jetpack is no longer a requirement?

-------------------

Closes #1253
